File: algoritmos/2_super_classe_animal/animais/andorinha.dart
import '../interfaces/acao_animal_interface.dart';
import '../tipos/ave.dart';

class Andorinha extends Ave implements AcaoAnimalInterface {
  String cor;

  Andorinha(
      this.cor, String sexo, int idade, double peso, int quantidadeDePatas)
      : super(sexo, idade, peso, quantidadeDePatas);

  @override
  void voar() {
    print('As andorinhas voam!');
  }
}

File: algoritmos/2/challenge-opp/Entities/Andorinha.cs
using challenge_opp.Interfaces;

namespace challenge_opp.Entities;
public sealed class Andorinha : Ave, IVoo
{
    public Andorinha(double peso, int quantidadeDePatas) : base(peso, quantidadeDePatas)
    {
    }

    public void Voar() =>
        Console.WriteLine("Andorinha voa");
}
